test/parallel/test-dns.js

```diff
@@ -62,6 +62,21 @@ assert(existing.length > 0);
   ]);
 }
 
+{
+  // Various invalidities, all of which should throw a clean error.
+  const invalidServers = [
+    ' ',
+    '\n',
+    '\0',
+    '1'.repeat(3 * 4),
+    ':'.repeat(100000),
+    '['.repeat(100000) // REDOS
+  ];
+  invalidServers.forEach((serv) => {
+    assert.throws(() => dns.setServers([serv]), /TypeError.*ERR_INVALID_IP_ADDRESS/, `Unexpected error thrown for ${serv}`);
```

Comment:
This line is too long, but I'm not sure how best to format it. I tried the obvious indentation but eslint complained that the TypeError ought to be indented 18 spaces (??).